src/utils/__tests__/changelog.test.ts

```diff
@@ -0,0 +1,504 @@
+      expect(changes).toMatchInlineSnapshot(`
+        "## Various fixes & improvements
+
+        - Upgraded the kernel (abcdef12)"
+      `);
+    });
+    it('should use short commit SHA for local commits w/o pull requests', async () => {
+      mockGetChangesSince.mockResolvedValueOnce([
+        {
+          hash: 'abcdef1234567890',
+          message: 'Upgraded the kernel',
+          pr: null,
+        },
+      ]);
+      makeCommitResponse([{ hash: 'abcdef1234567890' }]);
+      const changes = await generateChangesetFromGit(dummyGit, '1.0.0');
+      expect(changes).toMatchInlineSnapshot(`
+        "## Various fixes & improvements
+
+        - Upgraded the kernel (abcdef12)"
+      `);
+    });
+
+    it('should use pull request number when available locally', async () => {
+      mockGetChangesSince.mockResolvedValueOnce([
+        {
+          hash: 'abcdef1234567890',
+          message: 'Upgraded the kernel (#123)',
+          pr: '123',
+        },
+      ]);
+      makeCommitResponse([
+        {
+          hash: 'abcdef1234567890',
+        },
+      ]);
+      const changes = await generateChangesetFromGit(dummyGit, '1.0.0');
+      expect(changes).toMatchInlineSnapshot(`
+        "## Various fixes & improvements
+
+        - Upgraded the kernel (#123)"
+      `);
+    });
+
+    it('should use pull request number when available remotely', async () => {
+      mockGetChangesSince.mockResolvedValueOnce([
+        {
+          hash: 'abcdef1234567890',
+          message: 'Upgraded the kernel',
+          pr: null,
+        },
+      ]);
+      makeCommitResponse([{ hash: 'abcdef1234567890', pr: '123' }]);
+      const changes = await generateChangesetFromGit(dummyGit, '1.0.0');
+      expect(changes).toMatchInlineSnapshot(`
+        "## Various fixes & improvements
+
+        - Upgraded the kernel (#123)"
+      `);
+    });
+
+    it('should handle multiple commits properly', async () => {
+      mockGetChangesSince.mockResolvedValueOnce([
+        {
+          hash: 'abcdef1234567890',
+          message: 'Upgraded the kernel',
+          pr: null,
+        },
+        {
+          hash: 'bcdef1234567890a',
+          message: 'Upgraded the manifold (#123)',
+          pr: '123',
+        },
+        {
+          hash: 'cdef1234567890ab',
+          message: 'Refactored the crankshaft',
+          pr: null,
+        },
+      ]);
+      makeCommitResponse([
+        {
+          hash: 'bcdef1234567890a',
+          pr: '123',
+        },
+        {
+          hash: 'cdef1234567890ab',
+          pr: '456',
+        },
+      ]);
+      const changes = await generateChangesetFromGit(dummyGit, '1.0.0');
+      expect(changes).toMatchInlineSnapshot(`
+        "## Various fixes & improvements
+
+        - Upgraded the kernel (abcdef12)
+        - Upgraded the manifold (#123)
+        - Refactored the crankshaft (#456)"
+      `);
+    });
+  });
+  it('should handle group prs under milestones', async () => {
```

Comment:
This test isn't only testing group prs, but also group prs + commits without prs. This should be reflected in the naming.

```suggestion
  it('should handle group prs under milestones and individual commits', async () => {
```

We may also want to include prs that don't belong to any milestone (in a test where some prs belong to milestones).

Reply:
>We may also want to include prs that don't belong to any milestone (in a test where some prs belong to milestones).

We have that case above on lines 414 and 415? We can add a similar PR here but I don't see the added benefit/coverage.